### PR TITLE
feat(tests): migrate to circle 2

### LINKED ISSUE #6336: Update to Circle Ci 2 before Aug 31

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes #6336

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 2c1a0113e79aced7366a47850784dd82a50415db
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 2c1a0113e79aced7366a47850784dd82a50415db
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 15223e89c74876969269b56656c81e42002c336f
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 2c1a0113e79aced7366a47850784dd82a50415db
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 15223e89c74876969269b56656c81e42002c336f
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: dc223216b8174154d5650e6b96d1b3af141cc342
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 2c1a0113e79aced7366a47850784dd82a50415db
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 15223e89c74876969269b56656c81e42002c336f
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: dc223216b8174154d5650e6b96d1b3af141cc342
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: de9db58c0fb366e21808ab51bce702b11e9bc9c6
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 2c1a0113e79aced7366a47850784dd82a50415db
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 15223e89c74876969269b56656c81e42002c336f
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: dc223216b8174154d5650e6b96d1b3af141cc342
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: de9db58c0fb366e21808ab51bce702b11e9bc9c6
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 720685c9eacec8876a2022175f129847d862c379
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 2c1a0113e79aced7366a47850784dd82a50415db
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 15223e89c74876969269b56656c81e42002c336f
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: dc223216b8174154d5650e6b96d1b3af141cc342
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: de9db58c0fb366e21808ab51bce702b11e9bc9c6
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 720685c9eacec8876a2022175f129847d862c379
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: bc56fc2a4370e8930cd90059fd193503e9135039
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 2c1a0113e79aced7366a47850784dd82a50415db
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 15223e89c74876969269b56656c81e42002c336f
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: dc223216b8174154d5650e6b96d1b3af141cc342
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: de9db58c0fb366e21808ab51bce702b11e9bc9c6
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 720685c9eacec8876a2022175f129847d862c379
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: bc56fc2a4370e8930cd90059fd193503e9135039
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/mozilla/fxa-content-server/blob/master/CONTRIBUTING.md#git-commit-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by shane-tomlinson** [comment]
Oh my! The tests are all passing!

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+            - build
+            - test1:
+                requires:
+                    - build
+            - test2:
+                requires:
+                    - build
+            - test3:
+                requires:
+                - build
```
Comment (vbudhram) [inline .circleci/config.yml]: nit: indent off on the lines below?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+          - DISPLAY=:99
+
+        steps:
+        - checkout
+        - run:
+              command: |
+                  sudo apt-get install -y python-setuptools python-dev build-essential xvfb
+                  sudo easy_install pip
+                  sudo pip install mozdownload mozinstall
+                  mozdownload --version 58.0 --destination firefox.tar.bz2
```
Comment (vladikoff) [inline .circleci/config.yml]: filed a follow up for caching this: https://github.com/mozilla/fxa-content-server/issues/6414

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+              paths: ./*
+
+    test1:
+        environment:
+            CIRCLE_NODE_INDEX: 0
+            CIRCLE_NODE_TOTAL: 4
+
+        working_directory: ~/fxa
+        docker:
+            - image: circleci/node:8-stretch-browsers
```
Comment (vbudhram) [inline .circleci/config.yml]: Curious why this is different from the one in the build step?

Comment (vladikoff) [reply .circleci/config.yml]: This images comes with dependencies require for browser testing. See https://circleci.com/docs/2.0/browser-testing/#overview

File: .circleci/config.yml (hunk 2 above)
Comment (jrgm) [reply .circleci/config.yml]: why 58.0?

Comment (vladikoff) [reply .circleci/config.yml]: that's what we used in the old circle file

**Pull request merged**